File: backend/src/main/java/com/tpvgallega/backend/domain/model/Pedido.java
package com.tpvgallega.backend.domain.model;

import java.time.LocalDateTime;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
public class Pedido {

    private Long id;
    private int mesa;
    private List<LineaPedido> lineas;
    private EstadoPedido estado;
    private LocalDateTime fechaCreacion;
    private LocalDateTime fechaActualizacion;

    public boolean tieneLineasDeTipo(TipoProducto tipo) {
        return lineas != null && lineas.stream().anyMatch(linea -> linea.getTipoProducto() == tipo);
    }
}

File: frontend/src/types/pedido.ts
export type EstadoPedido = 'PENDIENTE' | 'PREPARANDO' | 'LISTO' | 'ENTREGADO' | 'CANCELADO';

export type TipoProducto = 'COMIDA' | 'BEBIDA';

export interface LineaPedido {
  id?: number;
  nombreProducto: string;
  cantidad: number;
  tipoProducto: TipoProducto;
  notas?: string;
}

export interface Pedido {
  id: number;
  mesa: number;
  estado: EstadoPedido;
  fechaCreacion: string;
  fechaActualizacion: string;
  lineas: LineaPedido[];
}

export interface CrearPedidoRequest {
  mesa: number;
  lineas: LineaPedido[];
}
